hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCommitArchiver.java

```diff
@@ -0,0 +1,273 @@
+
+        if (!commitTimeline.empty() && commitTimeline.countInstants() > maxCommitsToKeep) {
+            // Actually do the commits
+            return commitTimeline.getInstants().filter(s -> !savepoints.contains(s.getTimestamp()))
+                .limit(commitTimeline.countInstants() - minCommitsToKeep);
+        }
+        return Stream.empty();
+    }
+
+    private boolean deleteInstants(List<HoodieInstant> commitsToArchive) {
+        log.info("Deleting instant " + commitsToArchive);
+        HoodieTableMetaClient metaClient =
+            new HoodieTableMetaClient(fs, config.getBasePath(), true);
+
+        boolean success = true;
+        for (HoodieInstant commitToArchive : commitsToArchive) {
+            Path commitFile =
+                new Path(metaClient.getMetaPath(), commitToArchive.getFileName());
+            try {
+                if (fs.exists(commitFile)) {
+                    success &= fs.delete(commitFile, false);
+                    log.info("Archived and deleted instant file " + commitFile);
+                }
+            } catch (IOException e) {
+                throw new HoodieIOException("Failed to delete archived instant " + commitToArchive,
+                    e);
+            }
+        }
+        return success;
+    }
+
+    public void archive(List<HoodieInstant> commits) throws HoodieCommitException {
+
+        try {
+            HoodieTableMetaClient metaClient =
+                    new HoodieTableMetaClient(fs, config.getBasePath(), true);
+            HoodieTimeline commitTimeline =
+                    metaClient.getActiveTimeline().getAllCommitsTimeline().filterCompletedInstants();
+
+            Schema wrapperSchema = HoodieMetadataWrapper.getClassSchema();
+            log.info("Wrapper schema " + wrapperSchema.toString());
+            List<IndexedRecord> records = new ArrayList<>();
+            for (HoodieInstant hoodieInstant : commits) {
+                records.add(convertToAvroRecord(commitTimeline, hoodieInstant));
+            }
+            HoodieAvroDataBlock block = new HoodieAvroDataBlock(records, wrapperSchema);
+            writer.appendBlock(block);
+        } catch(Exception e) {
+            throw new HoodieCommitException("Failed to archive commits", e);
+        }
+    }
+
+    public Path getArchiveFilePath() {
+        return archiveFilePath;
+    }
+
+    private IndexedRecord convertToAvroRecord(HoodieTimeline commitTimeline, HoodieInstant hoodieInstant) throws IOException {
+        HoodieMetadataWrapper wrapper = new HoodieMetadataWrapper();
+        wrapper.setCommitTime(hoodieInstant.getTimestamp());
+        switch(hoodieInstant.getAction()) {
+            case HoodieTimeline.CLEAN_ACTION:{
+                wrapper.setHoodieCleanMetadata(AvroUtils.deserializeAvroMetadata(commitTimeline.getInstantDetails(hoodieInstant).get(), HoodieCleanMetadata.class));
+                wrapper.setActionType(ActionType.HoodieClean);
```

Comment:
Agreed, I wanted to keep the actiontype the same as how we refer to the payloads themselves. But on second thoughts, looking at actions in HoodieTimeline it makes sense to not prefix Hoodie.